cmake/modules/AddSwift.cmake

```diff
@@ -75,6 +75,41 @@ function(_set_target_prefix_and_suffix target kind sdk)
   endif()
 endfunction()
 
+function(_add_host_variant_swift_sanitizer_flags target)
+  if (LLVM_USE_SANITIZER)
+    if (LLVM_ON_UNIX)
+
+      if (LLVM_USE_SANITIZER STREQUAL "Address")
+        set(_Swift_SANITIZER_FLAGS "-sanitize=address")
+      elseif (LLVM_USE_SANITIZER STREQUAL "HWAddress")
+        # Not supported?
+      elseif (LLVM_USE_SANITIZER MATCHES "Memory(WithOrigins)?")
+        # Not supported
+        if(LLVM_USE_SANITIZER STREQUAL "MemoryWithOrigins")
+          # Not supported
+        endif()
+      elseif (LLVM_USE_SANITIZER STREQUAL "Undefined")
+        set(_Swift_SANITIZER_FLAGS "-sanitize=undefined")
+      elseif (LLVM_USE_SANITIZER STREQUAL "Thread")
+        set(_Swift_SANITIZER_FLAGS "-sanitize=thread")
+      elseif (LLVM_USE_SANITIZER STREQUAL "DataFlow")
+        # Not supported
+      elseif (LLVM_USE_SANITIZER STREQUAL "Address;Undefined" OR
+              LLVM_USE_SANITIZER STREQUAL "Undefined;Address")
+        set(_Swift_SANITIZER_FLAGS "-sanitize=address -sanitize=undefined")
```

Comment:
Using [`FIND`](https://cmake.org/cmake/help/latest/command/string.html#find) might be simpler compared to trying to match combinations using `STREQUAL`.

Reply:
Maybe. I just pulled the block from LLVM so that the logic matches.
Eventually, we should upstream it so the LLVM is assigning the flags appropriately, but this should get us through until that happens. I'd like to avoid deviating from what they're doing for now if at all possible.

Reply:
Didn't realize the context, sounds reasonable.